packages/components/progress-bar/src/progress-bar.tsx

```diff
@@ -0,0 +1,295 @@
+const getLabel = (label: NonNullable<TProgressBarProps['label']>) => {
+  if (typeof label === 'string') return label;
+  return label.hasOwnProperty('defaultMessage') ? (
+    <FormattedMessage {...label} />
+  ) : (
+    label
+  );
+};
+
+const ProgressBarLabel = (
+  props: Pick<
+    TProgressBarProps,
+    'label' | 'labelWidth' | 'isInverted' | 'height' | 'labelPosition'
+  >
+) => {
+  if (isNil(props.label)) return null;
+  return (
+    <Constraints.Horizontal max={props.labelWidth}>
+      <div
+        css={css`
+          min-height: ${heightPerScale[props.height ?? '20']};
+          ${(props.labelPosition === 'top' ||
+            props.labelPosition === 'bottom') &&
+          css`
+            text-align: center;
+          `}
+        `}
+      >
+        {props.height === '10' ? (
+          <Text.Detail tone={props.isInverted ? 'inverted' : undefined}>
+            {getLabel(props.label)}
+          </Text.Detail>
+        ) : (
+          <Text.Body
+            as="span"
+            tone={props.isInverted ? 'inverted' : undefined}
+            fontWeight="medium"
+          >
+            {getLabel(props.label)}
+          </Text.Body>
+        )}
+      </div>
+    </Constraints.Horizontal>
+  );
+};
+
+const progressPulse = keyframes`
+  0% {
+    background-position: 100% 0;
+  }
+  100% {
+    background-position: -100% 0;
+  }
+`;
+
+const Bar = (
+  props: Pick<
+    TProgressBarProps,
+    'progress' | 'height' | 'barWidth' | 'isInverted' | 'isAnimated'
+  >
+) => {
+  return (
+    <Constraints.Horizontal max={props.barWidth}>
+      <div
+        css={css`
+          background-color: ${props.isInverted
+            ? 'rgba(255, 255, 255, 0.4)'
+            : designTokens.colorNeutral90};
+          border-radius: ${designTokens.spacingL};
+          height: ${heightPerScale[props.height ?? '20']};
+          overflow: hidden;
+        `}
+        role="progressbar"
+        aria-valuenow={props.progress}
+        aria-label="Progress bar"
+        {...filterAriaAttributes(props)}
+      >
+        <div
+          css={css`
+            width: ${props.progress}%;
+            transition: width 500ms ease-in-out;
+            display: block;
+            height: ${heightPerScale[props.height ?? '20']};
+            background: ${props.isInverted
+              ? designTokens.colorSurface
+              : `linear-gradient(
+                to right,
+                #00E5CB,
+                ${designTokens.colorPrimary25},
+                #00E5CB
+              )`};
+            background-size: 200% 100%;
+            animation: ${props.isAnimated && !props.isInverted
+              ? css`
+                  ${progressPulse} 2s linear infinite
+                `
+              : 'none'};
+            border-radius: ${designTokens.spacingL};
+          `}
+        />
+      </div>
+    </Constraints.Horizontal>
+  );
+};
+
+const ProgressBar = (props: TProgressBarProps) => {
```

Comment:
Just a refactor idea so we don't repeat the `Bar` and `ProgressBarLabel` components that much:

```tsx
const ProgressBar = (props: TProgressBarProps) => {
  const BarWithProps = (
    <Bar
      progress={props.progress}
      barWidth={props.barWidth}
      isInverted={props.isInverted}
      height={props.height}
      isAnimated={props.isAnimated}
    />
  );

  const LabelWithWithProps = (
    <ProgressBarLabel
      label={props.label}
      labelWidth={props.labelWidth}
      labelPosition={props.labelPosition}
      isInverted={props.isInverted}
      height={props.height}
    />
  );

  switch (props.labelPosition) {
    case 'bottom':
      return (
        <SpacingsStack scale="m" alignItems="center">
          {BarWithProps}
          {LabelWithWithProps}
        </SpacingsStack>
      );
    case 'left':
      return (
        <SpacingsInline scale="m" alignItems="center">
          <div
            css={css`
              text-align: right;
            `}
          >
            <SpacingsInline justifyContent="center">
              {LabelWithWithProps}
            </SpacingsInline>
          </div>
          {BarWithProps}
        </SpacingsInline>
      );
    case 'right':
      return (
        <SpacingsInline scale="m" alignItems="center">
          {BarWithProps}
          <SpacingsInline justifyContent="center">
            {LabelWithWithProps}
          </SpacingsInline>
        </SpacingsInline>
      );
    case 'top':
    default:
      return (
        <SpacingsStack scale="m" alignItems="center">
          {LabelWithWithProps}
          {BarWithProps}
        </SpacingsStack>
      );
  }
};
```

Reply:
@tylermorrisford I kept thinking in this rendering logic and I ended up with an alternative solution which I try to showcase in this alternative PR with the addition of extracting styles to its own file.

Let me see what you think about it 🙏

Reply:
I like it! I will test your solution a bit more but should I close this?

Reply:
If you mean this PR, no. You can merge the other PR into this one or pick the bits you agree with.